### ISSUE TITLE
Communes are not always LAU2

### ISSUE THREAD

**Issue by brynpickering** [timtroendle/possibility-for-electricity-autarky#5]
The [dataset](https://ec.europa.eu/eurostat/web/gisco/geodata/reference-data/administrative-units-statistical-units/communes#communes13) you use to inform the 'LAU2'/'commune' spatial resolution is not strictly 2013 LAU2. For instance, the data for the UK is more correctly 'wards'. The IDs for these wards don't always match LAU2 and, in the case of Scotland and Northern Ireland, their boundaries don't match LAU2 either. I suspect that *all* the communes with `None` values in the `TRUE_COMM_` column are not what they seem, i.e.:

![image](https://user-images.githubusercontent.com/17178478/79545476-a4fa5c00-8088-11ea-9eea-ee6a30523925.png)

This isn't a problem for your paper's analysis, since you don't aggregate based on IDs, but rather by using spatial joins. I only noticed the issue when trying to match known electricity consumption in the UK, at the LAU1 resolution to data coming out of this. I thought would be easiest (ha!) to do it by aggregating using LAU2 -> LAU1 [correspondence tables](https://ec.europa.eu/eurostat/web/nuts/local-administrative-units)).

**Comment by timtroendle** [comment]
Thanks for pointing this out. The data is unfortunately very confusing. Codes and classification change often, and are somehow inconsistent. For example, I also use the [DEGURBA dataset](https://ec.europa.eu/eurostat/ramon/miscellaneous/index.cfm?TargetUrl=DSP_DEGURBA) for LAU 2013 data in the analysis, which matches mostly, but not always the codes. Back when I did the analysis, this was the only existing dataset available. I was always hoping eurostat would publish a newer dataset (based on the new classification in which there is only one LAU level left), checked regularly their website -- but nothing.

I looked it up once more just now, and indeed they published [a newer dataset](https://ec.europa.eu/eurostat/web/gisco/geodata/reference-data/administrative-units-statistical-units/lau#lau18) last June. So if there is reason to, one could update the analysis with LAU 2018 and NUTS 2016.

**Comment by brynpickering** [comment]
Yeah, I agree that it is a total mess of codes (just the UK has about 5 different ways of classifying areas of a similar size and their codes/boundaries change at different points in time to each other). The unified LAU from 2017 onwards would be possible, but it's a closer match to LAU1 than LAU2 (e.g. ~400 UK areas instead of ~10,000). Would this pose a problem, do you imagine?

**Comment by timtroendle** [comment]
No, that should be straightforward (as in: you update the URL, rerun the workflow, it breaks, and you start debugging 😜).

But seriously, the only thing that comes to my mind is the DEGURBA dataset, which wouldn't match anymore (which is necessary only for the analysis of the potentials, that you may not be interested in). And most probably the pre-processing of the LAU data would need to be updated, in particular the mapping of column names.

**Comment by brynpickering** [comment]
Yeah, I wouldn't expect it to be too much of a problem programmatically, but do we care about the super-high resolution of 2013 'LAU2' beyond the study you undertook in this paper?

RE DEGURBA, the latest correspondence tables include this classification, so updating to 2018 would work for LAU & DEGBURA, provided we're happy with the somewhat coarser resolution

**Comment by timtroendle** [comment]
Whether or not we care about the resolution depends on the use case. If we are talking about euro-calliope, I'd say coarser is better (and especially newer is better).

**Comment by brynpickering** [comment]
Agreed. I'll update the finest resolution to LAU2017/2018 after/if we make the repository split discussed in timtroendle/solar-and-wind-potentials#2
